### PR TITLE
Feat/prsd 1146 epc expiry check

### PR DESCRIPTION
**Pull request by JasminConterioSW**
## Ticket number

PRSD-1146

## Goal of change

Add the epc expiry check page

## Description of main change(s)

Add the epc expiry check page
image

## Anything you'd like to highlight to the reviewer?

N/A

## Checklist

Delete any that are not applicable, and add explanation below for any that are applicable but haven't been done

- [X] Screenshots of any UI changes have been added
- [X] Unit tests for new logic (e.g. new service methods) have been added
- [X] Single page integration tests have been added for any unhappy-flow UI features, e.g. validation errors
- [X] New journey steps have been added to the appropriate journey integration test(s)
- [X] A new journey integration test has been added for any new journeys
- [x] Test suite has been run in full locally and is passing
- [x] Branch has been rebased onto main and run locally, with everything working as expected (both for your new feature
  and any related functionality)

### CODE REVIEW

File: src/main/kotlin/uk/gov/communities/prsdb/webapp/forms/journeys/PropertyComplianceJourney.kt [hunk 1]
```diff
@@ -983,6 +981,37 @@ class PropertyComplianceJourney(
...
+                                        ),
+                                        RadiosButtonViewModel(
+                                            value = false,
+                                            valueStr = "no",
+                                            labelMsgKey = "forms.radios.option.no.label",
+                                        ),
+                                    ),
+                            ),
+                    ),
+                nextAction = { journeyData, _ -> epcExpiryCheckStepNextAction(journeyData) },
```
Comment (isobel-softwire) [inline src/main/kotlin/uk/gov/communities/prsdb/webapp/forms/journeys/PropertyComplianceJourney.kt:1012]: Minor but this is `filteredJourneyData` now

File: src/main/kotlin/uk/gov/communities/prsdb/webapp/forms/journeys/PropertyComplianceJourney.kt [hunk 2]
```diff
@@ -1277,6 +1306,18 @@ class PropertyComplianceJourney(
         return updateEpcDetailsInSessionAndRedirectToNextStep(epcLookupStep, filteredJourneyData, latestEpc, autoMatchedEpc = false)
     }
 
+    private fun epcExpiryCheckStepNextAction(filteredJourneyData: JourneyData): Pair<PropertyComplianceStepId?, Int?> =
+        if (filteredJourneyData.getEpcExpiryCheckTenancyStartedBeforeExpiry() == true) {
+            val sessionJourneyData = journeyDataService.getJourneyDataFromSession()
+            if (sessionJourneyData.getAcceptedEpcDetails()?.isEnergyRatingEOrBetter() == true) {
```
Comment (isobel-softwire) [inline src/main/kotlin/uk/gov/communities/prsdb/webapp/forms/journeys/PropertyComplianceJourney.kt:1312]: You should be able to call `getAcceptedEpcDetails` on the `filteredJourneyData` as non-step related values are always included.

File: src/main/kotlin/uk/gov/communities/prsdb/webapp/helpers/extensions/journeyExtensions/PropertyComplianceJourneyDataExtensions.kt [hunk 3]
```diff
@@ -212,6 +213,13 @@ class PropertyComplianceJourneyDataExtensions : JourneyDataExtensions() {
                 EicrExemptionReasonFormModel::exemptionReason.name,
             )
 
+        fun JourneyData.getEpcExpiryCheckTenancyStartedBeforeExpiry(): Boolean? =
```
Comment (isobel-softwire) [inline src/main/kotlin/uk/gov/communities/prsdb/webapp/helpers/extensions/journeyExtensions/PropertyComplianceJourneyDataExtensions.kt:216]: This method name is a little confusing. I think something like `getDidTenancyStartBeforeEpcExpiry` would make its purpose clearer.

File: src/main/resources/templates/forms/epcExpiryCheckForm.html [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<!--/*@thymesVar id="title" type="java.lang.String"*/-->
+<!--/*@thymesVar id="formModel" type="java.lang.Object"*/-->
+<!--/*@thymesVar id="backUrl" type="java.lang.String"*/-->
+<!--/*@thymesVar id="sectionHeaderInfo" type="uk.gov.communities.prsdb.webapp.models.viewModels.SectionHeaderViewModel"*/-->
+<!--/*@thymesVar id="fieldSetHeading" type="java.lang.String"*/-->
+<!--/*@thymesVar id="radioOptions" type="uk.gov.communities.prsdb.webapp.models.viewModels.formModels.RadiosViewModel"*/-->
+<!DOCTYPE html>
+<html id="form-content" th:replace="~{fragments/forms/questionPage :: questionPage(#{${title}}, ${formModel}, ~{::#form-content/content()}, ${backUrl}, ${sectionHeaderInfo})}">
+<th:block id="fieldset-content" th:replace="~{fragments/forms/fieldSet :: fieldSet(~{::#fieldset-content/content()}, 'hasExemption', #{${fieldSetHeading}}, null)}">
```
Comment (isobel-softwire) [inline src/main/resources/templates/forms/epcExpiryCheckForm.html:9]: Should 'hasExemption' be 'tenancyStartedBeforeExpiry'?

File: src/main/resources/templates/forms/epcExpiryCheckForm.html [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    <p class="govuk-body" th:text="#{forms.epcExpiryCheck.paragraph.one}">forms.epcExpiryCheck.paragraph.one</p>
+    <p class="govuk-body" th:text="#{forms.epcExpiryCheck.paragraph.two}">forms.epcExpiryCheck.paragraph.two</p>
+    <p class="govuk-body"><strong  th:text="#{forms.epcExpiryCheck.paragraph.three(${{expiryDate}})}">forms.epcExpiryCheck.paragraph.three</strong></p>
```
Comment (isobel-softwire) [inline src/main/resources/templates/forms/epcExpiryCheckForm.html:12]: Minor but I think there's an extra space after `<strong`

File: src/test/kotlin/uk/gov/communities/prsdb/webapp/helpers/extensions/journeyExtensions/PropertyComplianceJourneyDataExtensionsTests.kt [hunk 6]
```diff
@@ -660,6 +660,15 @@ class PropertyComplianceJourneyDataExtensionsTests {
         assertEquals(CURRENT_EPC_CERTIFICATE_NUMBER, retrievedCertificateNumber)
     }
 
+    @Test
+    fun `getEpcExpiryCheckTenancyStartedBeforeExpiry returns the submitted answer for the EpcExpiryCheck step`() {
```
Comment (isobel-softwire) [inline src/test/kotlin/uk/gov/communities/prsdb/webapp/helpers/extensions/journeyExtensions/PropertyComplianceJourneyDataExtensionsTests.kt:664]: Might be worth adding a test to assert that it returns null if there isn't an answer for the EpcExpiryCheck check

File: src/test/kotlin/uk/gov/communities/prsdb/webapp/helpers/extensions/journeyExtensions/PropertyComplianceJourneyDataExtensionsTests.kt [hunk 7]
```diff
@@ -735,47 +744,41 @@ class PropertyComplianceJourneyDataExtensionsTests {
             assertTrue(hasCompletedEpcTask)
         }
 
-        // TODO: PRSD-1146 - add check that this page was answered "Yes"
-        @Disabled
         @Test
         fun `returns true if EpcExpiryCheck was answered Yes and the energy rating is E or better`() {
```
Comment (isobel-softwire) [inline src/test/kotlin/uk/gov/communities/prsdb/webapp/helpers/extensions/journeyExtensions/PropertyComplianceJourneyDataExtensionsTests.kt:748]: It would be helpful if the `getHasCompletedEpcTask` tests mention the method in their names

File: src/test/kotlin/uk/gov/communities/prsdb/webapp/testHelpers/builders/JourneyDataBuilder.kt [hunk 8]
```diff
@@ -647,6 +647,12 @@ class JourneyDataBuilder(
         return this
     }
 
+    fun withEpcExpiryCheckStep(tenancyStartedBeforeExpiry: Boolean): JourneyDataBuilder {
+        journeyData[PropertyComplianceStepId.EpcExpiryCheck.urlPathSegment] =
+            mapOf("tenancyStartedBeforeExpiry" to tenancyStartedBeforeExpiry)
```
Comment (isobel-softwire) [inline src/test/kotlin/uk/gov/communities/prsdb/webapp/testHelpers/builders/JourneyDataBuilder.kt:652]: You could use `EpcExpiryCheckFormModel::tenancyStartedBeforeExpiry.name` here

**Review by isobel-softwire — APPROVED** [review]
LGTM! Left some minor thoughts
